deployment/bicep/backend.bicep

```diff
@@ -0,0 +1,433 @@
+//
+// APIM Construction Set
+// Backend Resource Group
+//
+// Contains:
+// - Storage Account for Azure Function Apps
+// - Application Service Plan for Azure Function Apps
+// - Azure Function App with Code Stack
+// - Azure Function App with Container
+//
+//  mbecker@microsoft.com, 2021
+//
+
+// Deploy as
+//
+// az deployment group create -f .\rgAPIMCSBackend.bicep -g <TargetResourceGroup>
+//
+// Sample
+// az deployment group create -f .\rgAPIMCSBackend.bicep -g rgAPIMCSBackend
+// *Target Resource Group must exist prior to deployment
+
+//
+// Parameters
+//
+// Azure Storage Account name: must be globally unique
+param storageAccounts_saapimcsbackend_name string = 'saapimcsbackend1'
+// Azure Application Service Plan name
+param serverfarms_appsvcplanAPIMCSBackend_name string = 'appsvcplanAPIMCSBackend'
+// Azure Function App name (Code Stack): must be globally unique
+param sites_funcappAPIMCSBackendMicroServiceA_name string = 'funcappAPIMCSBackendMicroServiceA1'
+// Azure Function App name (Container): must be globally unique
+param sites_funcappAPIMCSBackendMicroServiceB_name string = 'funcappAPIMCSBackendMicroServiceB1'
+
+//
+// Definitions
+//
+// Azure Storage Account
+resource storageAccounts_saapimcsbackend_name_resource 'Microsoft.Storage/storageAccounts@2021-04-01' = {
+  name: storageAccounts_saapimcsbackend_name
+  location: 'westeurope'
+  sku: {
+    name: 'Standard_LRS'
+    tier: 'Standard'
+  }
+  kind: 'StorageV2'
+  properties: {
+    minimumTlsVersion: 'TLS1_2'
+    allowBlobPublicAccess: true
+    allowSharedKeyAccess: true
+    networkAcls: {
+      bypass: 'AzureServices'
+      virtualNetworkRules: []
+      ipRules: []
+      defaultAction: 'Deny'
+    }
+    supportsHttpsTrafficOnly: true
+    encryption: {
+      services: {
+        file: {
+          keyType: 'Account'
+          enabled: true
+        }
+        blob: {
+          keyType: 'Account'
+          enabled: true
+        }
+      }
+      keySource: 'Microsoft.Storage'
+    }
+    accessTier: 'Hot'
+  }
+}
+
+// Azure Application Service Plan
+resource serverfarms_appsvcplanAPIMCSBackend_name_resource 'Microsoft.Web/serverfarms@2018-02-01' = {
+  name: serverfarms_appsvcplanAPIMCSBackend_name
+  location: 'West Europe'
+  sku: {
+    name: 'B1'
+    tier: 'Basic'
+    size: 'B1'
+    family: 'B'
+    capacity: 1
+  }
+  kind: 'linux'
+  properties: {
+    perSiteScaling: false
+    maximumElasticWorkerCount: 1
+    isSpot: false
+    freeOfferExpirationTime: '7/18/2021 9:36:43 PM'
```

Comment:
The free offer is over, not sure if we'd like to include this

Reply:
agree: commented out so far